[enterprise-4.6] GH#33682: Updating the disk encryption section (#34427)
openshift/openshift-docs: modules/installation-special-config-encrypt-disk-tang.adoc

[id="installation-special-config-encrypt-disk-tang_{context}"]
= Enabling Tang disk encryption

> Use this procedure to enable Tang mode disk encryption during an {product-title} installation.

Reviewer: ```suggestion
Use the following procedure to enable Tang mode disk encryption during an {product-title} installation.
```

Author: @sagidlow thank you for your detailed review and suggestions! I have now resolved this suggestion.

.Prerequisites

> * You downloaded the {product-title} installation program on your installation node.

Reviewer: ```suggestion
* You have downloaded the {product-title} installation program on your installation node.
```

Author: Resolved.

====
+
.Example output
[source,terminal]
----
The advertisement contains the following signing keys:

PLjNyRdGw03zlRoGjQYMahSZGu9 <1>
----
<1> The thumbprint of the exchange key.
+
When the `Do you wish to trust these keys? [ynYN]` prompt displays, type `Y`.
+
[NOTE]
====
{op-system-base} 8 provides Clevis version 15, which uses the SHA-1 hash algorithm to generate thumbprints. Some other distributions provide Clevis version 17 or later, which use the SHA-256 hash algorithm for thumbprints. You must use a Clevis version that uses SHA-1 to create the thumbprint, to prevent Clevis binding issues when you install {op-system-first} on your {product-title} cluster nodes.
====

. Create a Base64 encoded file, replacing the URL of the Tang server (`url`) and thumbprint (`thp`) you just generated:
+
[source,terminal]
----
$ (cat <<EOM
{
 "url": "http://tang.example.com:7500", <1>
 "thp": "PLjNyRdGw03zlRoGjQYMahSZGu9" <2>
}
EOM
) | base64 -w0
----
<1> Specify the URL of a Tang server. In this example, `tangd.socket` is listening on port `7500` on the Tang server.
<2> Specify the exchange key thumbprint, which was generated in a preceding step.
+
.Example output
[source,terminal]
----

> ewogInVybCI6ICJodHRwczovL3RhbmcuZXhhbXBsZS5jb20iLAogInRocCI6ICJaUk1leTFjR3cwN3psVExHYlhuUWFoUzBHdTAiCn0K

Reviewer 2: Per the above example, the output should be:
```
ewogInVybCI6ICJodHRwOi8vdGFuZy5leGFtcGxlLmNvbTo3NTAwIiwgCiAidGhwIjogIlBMak55UmRHdzAzemxSb0dqUVlNYWhTWkd1OSIgCn0K
```

Author: @jianlinliu thank you for your detailed review and suggestions. I have subsequently updated the Base64 encoded output throughout the file.

----

> . Create machine config files to encrypt the boot disks for the control plane or compute nodes using the Tang encryption mode. 

Reviewer 2: It is better to prompt user to generate k8s manifests file before injecting the machineconfig files.

Author: Thank you for highlighting this. Resolved.

      files:
      - contents:
          source: data:text/plain;base64,e30K
          source: data:text/plain;base64,ewogInVybCI6ICJodHRwczovL3RhbmcuZXhhbXBsZS5jb20iLAogInRocCI6ICJaUk1leTFjR3cwN3psVExHYlhuUWFoUzBHdTAiCn0K <1>
        mode: 420
        overwrite: true
        path: /etc/clevis.json
  kernelArguments:
    - rd.neednet=1 <2>
----
<1> Specify the Base64 encoded string that was generated in the preceding step.
<2> Add the `rd.neednet=1` kernel argument to bring the network up in the initramfs. This argument is required.

> . Make a backup copy of the YAML files. You should do this because the manifest files will be consumed when you create the Ignition config files.

Reviewer: ```suggestion
. Create a backup copy of the YAML files.  The original YAML files are consumed when you create the Ignition config files.
```

Author: Resolved.
